experimental/PBWDeformations/LieAlgebras/LinearLieAlgebra.jl

```diff
@@ -0,0 +1,117 @@
+@attributes mutable struct LinearLieAlgebra{C<:RingElement} <: LieAlgebra{C}
+  R::Ring
+  n::Int  # the n of the gl_n this embeds into
+  dim::Int
+  basis::Vector{MatElem{C}}
+  s::Vector{Symbol}
+
+  function LinearLieAlgebra{C}(
+    R::Ring, n::Int, basis::Vector{<:MatElem{C}}, s::Vector{Symbol}; cached::Bool=true
+  ) where {C<:RingElement}
+    return get_cached!(
+      LinearLieAlgebraDict, (R, n, basis, s), cached
+    ) do
+      @req all(b -> size(b) == (n, n), basis) "Invalid basis element dimensions."
+      @req length(s) == length(basis) "Invalid number of basis element names."
+      new{C}(R, n, length(basis), basis, s)
+    end::LinearLieAlgebra{C}
+  end
+end
+
+const LinearLieAlgebraDict = CacheDictType{
+  Tuple{Ring,Int,Vector{<:MatElem},Vector{Symbol}},LinearLieAlgebra
+}()
+
+struct LinearLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
+  parent::LinearLieAlgebra{C}
+  mat::MatElem{C}
```

Comment:
This is not a concrete type, so operations with `LinearLieAlgebraElem` will never be type stable.

A solution would be to add a second type parameter, say `D<:MatElem`, and use that here; then an inner constructor can be added to enforce it is always suitable, i.e. something like this (untested code, may contain syntax errors, but I hope the gist is clear):
```suggestion
  mat::D
  LinearLieAlgebraElem{C}(parent::LinearLieAlgebra{C}, mat::MatElem{C}) where C
    new{C,typeof(mat})(parent, mat)
  end
```

Reply:
The other solution would be a type stable getter function. Overhead will probably be not signifcant.

Reply:
Currently, there is 
```julia
@inline function Generic._matrix(x::LieAlgebraElem{C}) where {C<:RingElement}
  return (x.mat)::dense_matrix_type(C)
end
```
for this. This is similar as for AA.FPModule